File: study/lua/src/app/fight/states/StateAction.lua
--播一个动作的state
local State = require("app.fight.states.State")
local StateAction = class("StateAction", State)

local actionTime = 0.3

function StateAction:ctor(entity, action)
    StateAction.super.ctor(self, entity)
    self._action = action
    self._entity:setAction(self._action)
    self._during = 0
end

function StateAction:update(f)
    if self:isStart() then
		if self._during >= actionTime then
            self:onFinish()
		end 
		self._during = self._during + f
	end
end


return StateAction

File: study/ts/script/fight/state/StateAction.ts
import { State } from "./State";
import Entity from "../unit/Entity";

export class StateAction extends State {
    private actionTime: number = 0.3;
    private action: string;
    private during = 0;

    constructor(entity: Entity, action: string) {
        super(entity);
        this.action = action;
        this.entity.setAction(this.action);
        this.during = 0;
        this.cName = "StateAction";
    }

    public update(f) {
        if (this.isStart) {
            if (this.during >= this.actionTime) {
                this.onFinish();
            }
            this.during += f;
        }
    }
}